ride_share.rb

```diff
@@ -0,0 +1,191 @@
+#Developer's name = Tithvorlak Mok (Lak)
+#Project: Ride Share
+#Due Date: 02/10/2020
+
+#Data of Rides
+total_rides = {
+    DR0001: [
+        {
+            date: '3rd Feb 2016',
+            cost: 10,
+            rider_id: 'RD0003',
+            rating: 3
+        },
+        {
+            date: '3rd Feb 2016',
+            cost: 30,
+            rider_id: 'RD0015',
+            rating: 4
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 45,
+            rider_id: 'RD0003',
+            rating: 2
+        }
+    ],
+    DR0002:[
+        {
+            date:'3rd Feb 2016',
+            cost: 25,
+            rider_id: 'RD0073',
+            rating: 5
+        },
+        {
+            date: '4th Feb 2016',
+            cost: 15,
+            rider_id:'RD0013',
+            rating: 1
+        },
+        {
+            date: '5th Feb 2016',
+            cost:35,
+            rider_id:'RD0006',
+            rating:3
+        }
+    ],
+    DR0003: [
+        {
+            date:'4th Feb 2016',
+            cost: 5,
+            rider_id: 'RD0066',
+            rating: 5
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 50,
+            rider_id: 'RD0003',
+            rating: 2
+        }
+    ],
+    DR0004: [
+        {
+            date: '3rd Feb 2016',
+            cost: 5,
+            rider_id: 'RD0022',
+            rating: 5
+        },
+        {
+            date:'4th Feb 2016',
+            cost: 10,
+            rider_id: 'RD0022',
+            rating: 4
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 20,
+            rider_id: 'RD0003',
+            rating: 5
+        }
+    ]
+}
+
+########################################################
+
+# - the number of rides each driver has given
+puts "The number of rides each driver has given"
+total_rides.each do |each_driver, rides|
+    puts "Driver #{each_driver} has #{rides.length}"
+end
+
+puts # Leave extra space for readability in the console 
+# - the total amount of money each driver has made
+puts "The total amount of money each driver has made"
+def all_costs_each_driver(total_rides)
+    all_costs_each_driver_array = total_rides.map do |each_driver, rides|
+        costs_of_each_driver = rides.map do |each_ride| #costs_of_each_driver is an array of all costs for each driver
+            each_ride[:cost]
+        end      
+        {
+            id: each_driver,
+            cost: costs_of_each_driver
+        }
+    end
+    return all_costs_each_driver_array
+end
+
+sum_cost_each_driver = all_costs_each_driver(total_rides) #Use the method all_costs_each_driver(total_rides)
+sum_cost_each_driver.each do |driver|
+    puts "Driver #{driver[:id]} has made $#{driver[:cost].sum} "
+end   
+
+puts # Leave extra space for readability in the console 
+# - the average rating for each driver
+puts "The average rating for each driver"
+def average_rating(total_rides)
```

Comment:
`total_rides` implies a single number, a better name might be `all_rides` which implies a list.